### PR TITLE
Remove ewasm target and rename usable parts for evm target

### PR DESCRIPTION
**Pull request by seanyoung**
The Ethereum Foundation has stopped funding for ewasm work, so it is
unlikely to happen in the current form. In addition, burrow has been end
of life'd by Hyperledger. So, the ewasm target is no longer useful.

We would love to have an evm target at some point in the future. Many
parts of the ewasm target is re-usable for evm, so we rename ewasm to
evm and remove the parts which are irrelevant for evm.

Note that the new evm target is good enough for the language server, so
the language server can be started in evm mode:

```
solang language-server --target evm
```

### CODE REVIEW

File: docs/language/builtins.rst
Comment (LucasSte) [inline docs/language/builtins.rst:262]: I should mention Borsh encoding here.

File: tests/codegen_testcases/yul/ewasm_builtin.sol
Comment (LucasSte) [inline tests/codegen_testcases/yul/ewasm_builtin.sol:6]: If we are still maintaining EVM for the language server, I suggest that this file remain as a test for the language server, verifying the correctness of the AST for all these builtins.

In addition, some of the builtins listed here are available for other targets (like `gas()`, `address()`, `caller()`, and others). They still need to be tested for code generation, but for another target.

File: tests/contract.rs
Comment (LucasSte) [inline tests/contract.rs:109]: Shouldn't we use `unimplemented!()`?

File: tests/ewasm_tests/yul.rs
Comment (LucasSte) [inline tests/ewasm_tests/yul.rs:9]: The builtins tested here are not tested elsewhere. There need to be tests for them on another target, depending on their compatibility.

**Review by LucasSte — CHANGES_REQUESTED** [review]

File: docs/language/builtins.rst
Comment (seanyoung) [reply docs/language/builtins.rst:262]: Well, we haven't moved to borsh encoding yet. We should mention it once we are done.

File: tests/contract.rs
Comment (seanyoung) [reply tests/contract.rs:109]: We don't want the test to fail.

File: tests/ewasm_tests/yul.rs
Comment (seanyoung) [reply tests/ewasm_tests/yul.rs:9]: We can't do runtime tests for the builtins anymore for ewasm/evm because we don't have codegen. So, I've added the tests to `tests/evm.rs`

File: tests/codegen_testcases/yul/ewasm_builtin.sol
Comment (seanyoung) [reply tests/codegen_testcases/yul/ewasm_builtin.sol:6]: So the reason I removed it is because the option `--target ewasm` was gone, and I had not added a new option `--target evm`. 

Now I've changed it so that `solang --target evm` is supported, but it is not documented in `solang compile --help` nor in the documentation,

**Review by xermicus — APPROVED** [review]

File: tests/ewasm_tests/yul.rs
Comment (LucasSte) [reply tests/ewasm_tests/yul.rs:9]: This is the only place in Solang where I test `address`, `selfbalance`, `caller`, `callvalue`, and `extcodesize`. By removing these runtime tests, there will be no test for them at all.

`address`, `caller`, `callvalue` and `selfbalance` are available for both Solana and Substrate. Instead of deleting the test, please, move it to a Solana or a Substrate runtime test. Only `extcodesize` is not available for those targets.

Comment (seanyoung) [reply tests/ewasm_tests/yul.rs:9]: I had missed that, thanks for pointing that out.

**Comment by seanyoung** [comment]
sorry @xermicus I did not mean to for a review again, you've already approved it

File: tests/codegen_testcases/yul/evm_builtin.sol
Comment (LucasSte) [inline tests/codegen_testcases/yul/evm_builtin.sol:3]: We don't have code generation for EVM, so how is this file tested?

File: tests/undefined_variable_detection.rs
Comment (LucasSte) [inline tests/undefined_variable_detection.rs:13]: I think undefined variable detection depends on codegen. Will this test work even if we use `Target::EVM`?

File: tests/unused_variable_detection.rs
Comment (LucasSte) [inline tests/unused_variable_detection.rs:12]: Same as above.

**Review by LucasSte — COMMENTED** [review]
I found no problem with the code. I just have some questions before granting approval.

File: tests/codegen_testcases/yul/evm_builtin.sol
Comment (seanyoung) [reply tests/codegen_testcases/yul/evm_builtin.sol:3]: We have codegen for evm (as in `src/codegen/..`), just emit.

File: tests/undefined_variable_detection.rs
Comment (seanyoung) [reply tests/undefined_variable_detection.rs:13]: Same as above. We're keeping the code `src/codegen/...`.

**Review by LucasSte — APPROVED** [review]

**Pull request merged**
